server/src/Handler/Item.hs

```diff
@@ -79,17 +85,21 @@ validateItemPrice price =
         then Left "Price should be above 0"
         else Right price
 
-itemForm :: UserId -> Maybe Item -> Form Item
+itemForm :: UserId -> Maybe Item -> Form (Item, (FileInfo, UTCTime))
 itemForm userId mitem =
     renderSematnicUiDivs $
-    Item <$> areq textField "Name" (itemName <$> mitem) <*>
-    areq
-        (selectField companies)
-        (selectSettings "Company")
-        (itemCompany <$> mitem) <*>
-    areq priceField "Price" (itemPrice <$> mitem) <*>
-    lift (liftIO getCurrentTime) <*>
-    pure userId
+    (,) <$>
+    (Item <$> areq textField "Name" (itemName <$> mitem) <*>
+     areq
+         (selectField companies)
+         (selectSettings "Company")
+         (itemCompany <$> mitem) <*>
+     areq priceField "Price" (itemPrice <$> mitem) <*>
+     -- @todo: Why doesn't `map itemPdfFile mitem` work here?
+     pure (maybe Nothing itemPdfFile mitem) <*>
```

Comment:
So we have `mitem :: Maybe Item` and `itemPdfFile :: Item -> Maybe (Key PdfFile)`—is that right?

If that's the case, then `mitem >>= itemPdfFile` will do the trick. If `itemPdfFile` didn't return a `Maybe`, then `fmap` would work.